opencti-platform/opencti-graphql/src/graphql/sseMiddleware.js

```diff
@@ -132,7 +135,99 @@ const authenticate = async (req, res, next) => {
+  const capabilityControl = (s) => s.name === BYPASS || s.name === STREAMAPI;
+  return R.find(capabilityControl, user.capabilities || []) !== undefined;
+};
+
+const computeUserAndCollection = async (req, res, { context, user, id }) => {
+  const collections = await getEntitiesFromCache(context, user, ENTITY_TYPE_STREAM_COLLECTION);
+  const collection = collections.find((c) => c.id === id);
+
+  // if bypass, let's go
+  if (user.capabilities.findIndex(({ name }) => name === BYPASS) > -1 && collection?.public) {
+    return { streamFilters: JSON.parse(collection.filters ?? ''), collection };
+  }
+
+  // Build filters
+  let streamFilters = {};
+  if (id !== DEFAULT_LIVE_STREAM) {
+    if (!collection) {
+      res.statusMessage = 'This live stream doesnt exists';
+      res.status(404).end();
+      return { error: 'This live stream doesnt exists' };
+    }
+    if (!collection.public) {
+      // We should compare user.groups and stream.groups
+      const userGroups = await batchGroups(context, user, user.id, { batched: false, paginate: false });
+      const collectionGroups = await streamCollectionGroups(context, user, collection);
+      if (collectionGroups.length > 0) {
+        // User must have one of the collection groups
+        const collectionGroupIds = collectionGroups.map((g) => g.id);
+        const userGroupIds = userGroups.map((g) => g.id);
+        if (!collectionGroupIds.some((c) => userGroupIds.includes(c))) {
+          res.statusMessage = 'You need to have access granted for this live stream';
+          res.status(401).end();
+          return { error: 'You need to have access granted for this live stream' };
+        }
+      }
+    }
+    streamFilters = JSON.parse(collection.filters);
+  }
+  // Check rights
+  if (!isUserGlobalCapabilityGranted(user)) {
+    // Access to the global live stream
+    res.statusMessage = 'You are not authorized, please check your credentials';
+    res.status(401).end();
+    return { error: 'You are not authorized, please check your credentials' };
+  }
+  // If no marking part of filtering are accessible for the user, return
+  // It's better to prevent connection instead of having no events accessible
+  if (streamFilters.markedBy) {
+    const userMarkings = (user.allowed_marking || []).map((m) => m.internal_id);
+    const filterMarkings = (streamFilters.markedBy || []).map((m) => m.id);
+    const isUserHaveAccess = filterMarkings.some((m) => userMarkings.includes(m));
+    if (!isUserHaveAccess) {
+      res.statusMessage = 'You need to have access to specific markings for this live stream';
+      res.status(401).end();
+      return { error: 'You need to have access to specific markings for this live stream' };
+    }
+  }
+
+  return { streamFilters, collection };
+};
+
+const authenticateForPublic = async (req, res, next) => {
+  const context = executionContext('stream_authenticate');
+  const auth = await authenticateUserFromRequest(context, req, res);
+  if (auth) {
```

Comment:
Looks like could be done only once with a simple ternary.



    const user = auth ?? SYSTEM_USER;
    req.userId = user.id;
    req.user = user;
    req.capabilities = user.capabilities;
    req.allowed_marking = user.allowed_marking;
    req.expirationTime = utcDate().add(1, 'days').toDate();

Reply:
Done